jujugui/static/gui/src/app/app.js

```diff
@@ -1889,6 +1893,39 @@ YUI.add('juju-gui', function(Y) {
     },
 
     /**
+      Display post deployment help.
+
+      @param {String} id The entity ID of the charm or bundle.
+      @param {String} displayName The display name of the charm or bundle.
+      @param {Array} files An array of the charm or bundles files.
+    */
+    _displayPostDeployment: function(id, displayName, files) {
+      this._clearPostDeployment();
+      const charmstore = this.get('charmstore');
+
+      const entityUrl = (id) => {
+        try {
+          return window.jujulib.URL.fromString(id);
+        } catch (_) {
+          return window.jujulib.URL.fromLegacyString(id);
+        }
+      };
+
+      ReactDOM.render(
+        <window.juju.components.PostDeployment
+          changeState={this.state.changeState.bind(this.state)}
+          closePostDeployment={this._clearPostDeployment.bind(this)}
+          entityId={id}
```

Comment:
Maybe this function itself could accept an URL rather than a id string. Ideally, ids are only in use in the edges of the app, where the app interacts with external systems, while internally charm/bundles are always referenced by their URL object. I know the GUI uses id strings all over the place, but that's tech debt.